xds/src/test/java/io/grpc/xds/CsdsServiceTest.java

```diff
@@ -125,12 +122,12 @@ public void fetchClientConfig_invalidArgument() {
       }
     }
 
-    /** Unexpected exceptions translated to internal error status. */
+    /** Unexpected exceptions translated to Status.Code.INTERNAL. */
     @Test
     public void fetchClientConfig_unexpectedException() {
-      XdsClient throwingXdsClient = new XdsClient() {
+      XdsClient throwingXdsClient = new FakeXdsClient() {
         @Override
-        Map<String, ResourceMetadata> getSubscribedResourcesMetadata(ResourceType type) {
+        Map<ResourceType, Map<String, ResourceMetadata>> getSubscribedResourcesMetadataSnapshot() {
```

Comment:
Note: this was needed because I discovered this test to be false negative, see PR description:

> Fix false negative in fetchClientConfig_unexpectedException,
> where expected exception  was actually caused by missing
> bootstrap data, not by throwing error from the local 
> class implementation.